src/Hl7.Fhir.Specification/Validation/TypeRefValidationExtensions.cs

```diff
@@ -156,16 +160,22 @@ internal static OperationOutcome ValidateResourceReference(this Validator valida
                 // References within the instance are dealt with within the same validator,
                 // references to external entities will operate within a new instance of a validator (and hence a new tracking context).
                 // In both cases, the outcome is included in the result.
-                OperationOutcome childResult;
-
+                OperationOutcome childResult = new OperationOutcome();
+                
                 if (encounteredKind != ElementDefinition.AggregationMode.Referenced)
                 {
-                    childResult = validator.Validate(referencedResource, typeRef.TargetProfile, statedProfiles: null, statedCanonicals: null);
+                    var validatedProfiles = referencedResource is ScopedNodeWrapper sn ? sn.ValidatedProfiles : null;
```

Comment:
I do not understand this bit, maybe you can explain this on a call?

Reply:
referencedResource could in theory be also ScopedNode (actually anything that inherits form BaseScopedNode). I don't think it will ever be anything other than ScopedNodeWrapper, but I still wanted to be sure. 

Even so, the `: null` is definitely wrong given the next line of code. I'll change it.